server.js

```diff
@@ -0,0 +1,96 @@
+const EventEmitter = require('events');
+const express = require('express');
+const { json } = require('body-parser');
+
+class OompaServer extends EventEmitter {
+  constructor(appSchema,
+              healthcheck=(() => Promise.resolve(true)),
+              middlewares=[]) {
+    super();
+    this._middlewareChain = [];
+    this._healthcheck = healthcheck;
+    this._appSchema = appSchema;
+    this.app = this._getBaseApp(middlewares);
+    this._api = express.Router();
+    this._setupApi(appSchema);
+  }
+
+  listen(...args) {
+    this._apiRoute();
+    this._server = this.app.listen(...args);
+    return this._server;
+  }
+
+  close(...args) {
+    return this._server.close(...args);
+  }
+
+  api(middleware) {
+    this._middlewareChain.push(middleware);
+    return this;
+  }
+
+  _getBaseApp(middlewares) {
+    const app = express();
+    app.use(json());
+    for (const mid of middlewares) {
+      app.use(mid);
+    }
+    app.get('/healthcheck', (req, res) => {
+      return this._healthcheck()
+        .then(val => res.json(val))
+        .catch(err => res.status(500).json(err));
```

Comment:
Let's do `{ ok: true, data: val }` in the good case, `{ ok: false, error: err }` in the bad case.